renderer/src/common/components/settings/tabs/logs-tab.tsx

```diff
@@ -0,0 +1,114 @@
+import { Button } from '../../ui/button'
+import { useState, useEffect } from 'react'
+import { Copy, Check, Download } from 'lucide-react'
+import { delay } from '../../../../../../utils/delay'
+import log from 'electron-log/renderer'
+import { toast } from 'sonner'
+
+const LOG_PATHS = {
+  darwin: '~/Library/Logs/ToolHive/main.log',
+  win32: '%USERPROFILE%\\AppData\\Roaming\\ToolHive\\logs\\main.log',
+  linux: '~/.config/ToolHive/logs/main.log',
+} as const
+
+export function LogsTab() {
+  const [copied, setCopied] = useState<boolean>(false)
+  const [isDownloading, setIsDownloading] = useState<boolean>(false)
+  const [logPath, setLogPath] = useState<string>('')
+  const platform = window.electronAPI.platform
+
+  useEffect(() => {
+    const fetchLogContent = async () => {
+      try {
+        const logPath = await window.electronAPI.getMainLogContent()
+        setLogPath(logPath)
+      } catch {
+        const logPath = LOG_PATHS[platform as keyof typeof LOG_PATHS] || ''
+        setLogPath(logPath)
+      }
+    }
+    fetchLogContent()
+  }, [platform])
+
+  const handleCopyLogPath = async () => {
+    try {
+      await navigator.clipboard.writeText(logPath)
+      setCopied(true)
+      delay(2000).then(() => setCopied(false))
+    } catch (error) {
+      log.error('Failed to copy to clipboard:', error)
+    }
+  }
+
+  const handleDownloadLog = async () => {
```

Comment:
I think this would merit it's own function for readability/separation of concerns

I guess here I would want to see sth like 

```tsx
setIsDownloading(true)
try {
   downloadFile("toolhive-log.txt", getLogContent())
} catch (error) {
   // same logs
} finally {
   setIsDownloading(false)
}
```



but I think even the transition features from react could be leveraged to reduce the manual logic here?


```tsx
  const [isDownloading, startDownload] = useTransition();

// ... etc

   const handleDownloadLog = () => {
      startDownload(() => {
         downloadFile("toolhive-log.txt", getLogContent())
      })
   }
```

Reply:
I guess a hook would makes sense

Reply:
yup, also now that I'm thinking about it, I think that the interactive parts of the page could be extracted to a `<LogDownloader />` component as well

Reply:
I created an hook con root folder, in this way let's see if there will be other use case for a specific LogDownloader component, let's start with the hook that is totally makes sense in terms of readability